starters/express-typescript-typeorm-postgres/src/utils/log-helper.ts

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+const allowedLogLevels = process.env.ALLOWED_LOG_LEVELS || '[]';
+
+type LogLevel = 'INFO' | 'DEBUG' | 'WARN' | 'ERROR';
+
+let LOG_LEVELS: LogLevel[] = ['DEBUG', 'INFO', 'WARN', 'ERROR'];
+try {
+  const parsedLogLevels = JSON.parse(allowedLogLevels);
+  if (parsedLogLevels.length !== 0) {
+    LOG_LEVELS = parsedLogLevels;
+  }
+} catch (e) {
+  console.warn(
+    `[WARN] Could not parse ALLOWED_LOG_LEVELS env variable, logger defaults to all levels`,
+    e
+  );
+}
+
+class Logger {
+  private enabledLevels: Set<LogLevel>;
+
+  constructor(allowedLogLevels: LogLevel[]) {
+    this.enabledLevels = new Set(allowedLogLevels);
+    this.log(`Initialising logger with allowed levels: ${allowedLogLevels.join(', ')}`);
+  }
+
+  log(message?: any, ...optionalParams: any[]): void {
+    if (this.enabledLevels.has('INFO')) {
+      if (typeof message === 'string') {
+        console.log(`[INFO] ${message}`, ...optionalParams);
+      } else {
+        console.log(`[INFO]`, message, ...optionalParams);
+      }
+    }
+  }
+
+  info(message?: any, ...optionalParams: any[]): void {
+    this.log(message, ...optionalParams);
+  }
+
+  debug(message?: any, ...optionalParams: any[]): void {
+    if (this.enabledLevels.has('DEBUG')) {
+      if (typeof message === 'string') {
+        console.debug(`[DEBUG] ${message}`, ...optionalParams);
+      } else {
+        console.debug(`[DEBUG]`, message, ...optionalParams);
+      }
+    }
+  }
+
+  warn(message?: any, ...optionalParams: any[]): void {
```

Comment:
These are the types of the console methods, we only have a wrapper around the console, therefore, these methods should have the same types as the console methods.

Reply:
Because it is a log helper which allows setting the log levels and uses console under the hood. This way we import the helper which has methods that developers are used to, and it adds extra value when looking for log entries in a tool like cloudWatch

Reply:
I really want these methods to have the exact same type as the console methods, and I linked before the official types of the console methods.

Reply:
`any` turns off  type checking, but `unknown` does not.

I am thinking hard, but I can't think of an instance where having type `unknown` would behave any differently that having `any` except it would allow us to stay in `strict` mode for TS.

Can you think of anything that may possibly be different or make something weird if we used `unknown`?

Reply:
I implemented your unknown approach, thanks for the suggestion. :)